core/store/models/log_events.go

```diff
@@ -85,55 +86,75 @@ var topicFactoryMap = map[common.Hash]logRequestParser{
 	RunLogTopic20190207withoutIndexes:         parseRunLog20190207withoutIndexes{},
 }
 
-// TopicFiltersForRunLog generates the two variations of RunLog IDs that could
-// possibly be entered on a RunLog or a ServiceAgreementExecutionLog. There is the ID,
-// hex encoded and the ID zero padded.
-func TopicFiltersForRunLog(logTopics []common.Hash, jobID *ID) [][]common.Hash {
-	return [][]common.Hash{logTopics, {IDToTopic(jobID), IDToHexTopic(jobID)}}
+// LogBasedChainlinkJobInitiators are initiators which kick off a log-specified
+// chainlink job when an appropriate ethereum log is received.
+// (InitiatorFluxMonitor kicks off a job when a log is emitted, but the JobID is
+// not taken from the log)
+var LogBasedChainlinkJobInitiators = []string{InitiatorRunLog, InitiatorEthLog,
+	InitiatorServiceAgreementExecutionLog}
+
+// topicsForInitiatorsWhichRequireJobSpecTopic are the log topics which kick off
+// a user job with the given type of initiator. If chainlink has any jobs with
+// these initiators, it subscribes on startup to logs which match both these
+// topics and some representation of the job spec ID.
+var TopicsForInitiatorsWhichRequireJobSpecIDTopic = map[string][]common.Hash{
+	InitiatorRunLog: {RunLogTopic20190207withoutIndexes,
+		RunLogTopic20190123withFullfillmentParams, RunLogTopic0original},
+	InitiatorServiceAgreementExecutionLog: {ServiceAgreementExecutionLogTopic},
+}
+
+// initiationRequiresJobSpecId is true if jobs initiated by the given
+// initiatiatorType require that their initiating logs match their JobSpecIDs.
+func initiationRequiresJobSpecID(initiatorType string) bool {
+	_, ok := TopicsForInitiatorsWhichRequireJobSpecIDTopic[initiatorType]
+	return ok
+}
+
+// jobSpecIDTopics lists the ways jsID could be represented as a log topic. This
+// allows log subscriptions to respond to all possible representations.
```

Comment:
maybe? // jobSpecIDTopics lists the ways its ID could be represented as a log topic. This

Reply:
Changed it to `...the ways jobSpecID could be...`, and changed the variable name accordingly.